### ISSUE TITLE
Empty selection set produced with `@defer`'d query

### ISSUE THREAD

**Issue by abernix** [apollographql/federation#2123]
It seems that even with an otherwise valid query plan using `@defer`, the `prettyFormatQueryPlan` function will throw under particular (unknown, as of yet) circumstances.

This can be reproduced with [this schema](https://raw.githubusercontent.com/apollographql/supergraph-demo-fed2/main/supergraph.graphql) and the following operation on the latest 2.1.1 release:

```graphql
fragment MyFragment on DeliveryEstimates {
    estimatedDelivery
    fastestDelivery
}
query Dimensions {
  allProducts {
    delivery {
      ...MyFragment @defer
    }
    sku
    id
  }
}
```

**Comment by abernix** [comment]
Note: we now believe this might actually be defer query plan generation bug, as we later observed (after guarding against the printing bug) that `subselection` on the query plan was invalid GraphQL:

<img width="849" alt="image" src="https://user-images.githubusercontent.com/841294/187975388-9f58ba46-dc7b-4cff-867c-815d756474d6.png">

**Comment by pcmanus** [comment]
So yeah, this isn't really an issue with `prettyFormatQueryPlan`, it is one with the plan generated itself. Pushed fix at #2125. Explanation on what was wrong there.

**Comment by abernix** [comment]
I'll try to track this landing in the Router here: https://github.com/apollographql/router/issues/1698
